preprocessy/handlenullvalues/_main.py

```diff
@@ -23,20 +23,28 @@ class NullValuesHandler:
     Public Methods
     --------------
 
-    execute() : Main function that performs the operations on supplied dataframe and returns a new dataframe
+    execute(params) : Main function that performs the operations on supplied dataframe and returns a new dataframe
 
     """
 
-    def __init__(self, df=None, drop=None, fill_missing=None, fill_values=None):
-        self.df = df
-        self.drop = drop
-        self.fill_missing = fill_missing
-        self.fill_values = fill_values
-        self.__validate_input()
-        self.new_df = None
-        self.final_df = None
+    def __init__(self):
+        pass
+
+    def __validate_input(self, params):
+        if "df" not in params.keys():
+            params["df"] = None
+        if "drop" not in params.keys():
+            params["drop"] = None
+        if "fill_missing" not in params.keys():
+            params["fill_missing"] = None
+        if "fill_values" not in params.keys():
+            params["fill_values"] = None
+
+        self.df = params["df"]
+        self.drop = params["drop"]
+        self.fill_missing = params["fill_missing"]
+        self.fill_values = params["fill_values"]
```

Comment:
Move this to execute and revert `__validate_input` to how it was before.

Reply:
Okay, will do this.